src/Uno.UI.RuntimeTests/MUX/Microsoft_UI_Xaml_Controls/TreeViewTests.cs

```diff
@@ -0,0 +1,554 @@
+				Verify.AreEqual(treeView.RootNodes[0].Content as TreeViewItemSource, items[0]);
+			});
+		}
+
+		[TestMethod]
+		public void TreeViewItemsSourceUpdateTest()
+		{
+			RunOnUIThread.Execute(() =>
+			{
+				var treeView = new TreeView();
+				var items = CreateTreeViewItemsSource();
+				treeView.ItemsSource = items;
+
+				// Insert
+				var newItem = new TreeViewItemSource() { Content = "newItem" };
+				items.Add(newItem);
+				Verify.AreEqual(treeView.RootNodes.Count, 3);
+				var itemFromNode = treeView.RootNodes[2].Content as TreeViewItemSource;
+				Verify.AreEqual(newItem.Content, itemFromNode.Content);
+
+				// Remove
+				items.Remove(newItem);
+				Verify.AreEqual(treeView.RootNodes.Count, 2);
+
+				// Replace
+				var item3 = new TreeViewItemSource() { Content = "3" };
+				items[1] = item3;
+				itemFromNode = treeView.RootNodes[1].Content as TreeViewItemSource;
+				Verify.AreEqual(item3.Content, itemFromNode.Content);
+
+				// Clear
+				items.Clear();
+				Verify.AreEqual(treeView.RootNodes.Count, 0);
+
+			});
+		}
+
+		[TestMethod]
+		public void TreeViewNodeStringableTest()
+		{
+			RunOnUIThread.Execute(() =>
+			{
+				var node = new TreeViewNode() { Content = "Node" };
+				Verify.AreEqual(node.Content, node.ToString());
+
+				// Test inherited type
+				var node2 = new TreeViewNode2() { Content = "Inherited from TreeViewNode" };
+				Verify.AreEqual(node2.Content, node2.ToString());
+			});
+		}
+
+		[TestMethod]
+		public void TreeViewPendingSelectedNodesTest()
+		{
+			RunOnUIThread.Execute(() =>
+			{
+				var treeView = new TreeView();
+				treeView.SelectionMode = TreeViewSelectionMode.Multiple;
+
+				var node1 = new TreeViewNode() { Content = "Node1" };
+				var node2 = new TreeViewNode() { Content = "Node2" };
+				treeView.RootNodes.Add(node1);
+				treeView.RootNodes.Add(node2);
+				treeView.SelectedNodes.Add(node1);
+
+				Content = treeView;
+				Content.UpdateLayout();
+				Verify.AreEqual(true, IsMultiSelectCheckBoxChecked(treeView, node1));
+				Verify.AreEqual(false, IsMultiSelectCheckBoxChecked(treeView, node2));
+			});
+		}
+
+		//[TestMethod]
+		//public void VerifyVisualTree()
+		//{
+		//	TreeView treeView = null;
+		//	RunOnUIThread.Execute(() =>
+		//	{
+		//		treeView = new TreeView() { Width = 400, Height = 400 };
+		//		var node1 = new TreeViewNode() { Content = "Node1" };
+		//		treeView.RootNodes.Add(node1);
+		//	});
+		//	TestUtilities.SetAsVisualTreeRoot(treeView);
+
+		//	VisualTreeTestHelper.VerifyVisualTree(root: treeView, masterFilePrefix: "TreeView");
+		//}
+
+		private bool IsMultiSelectCheckBoxChecked(TreeView tree, TreeViewNode node)
+		{
+			var treeViewItem = tree.ContainerFromNode(node) as TreeViewItem;
+			var checkBox = FindVisualChildByName(treeViewItem, "MultiSelectCheckBox") as CheckBox;
+			return checkBox.IsChecked == true;
+		}
+
+		public static DependencyObject FindVisualChildByName(FrameworkElement parent, string name)
+		{
+			if (parent.Name == name)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this method to add validation of parameter 'parent' before using it.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)